sdk/python/feast/lineage/registry_lineage.py

```diff
@@ -0,0 +1,388 @@
+"""
+Registry lineage generation for Feast objects.
+
+This module provides functionality to generate relationship graphs between
+Feast objects (entities, feature views, data sources, feature services)
+for lineage visualization.
+"""
+
+from dataclasses import dataclass
+from enum import Enum
+from typing import Dict, List, Tuple
+
+from feast.protos.feast.core.Registry_pb2 import Registry
+
+
+class FeastObjectType(Enum):
```

Comment:
Does transform belong as its own object? I'm not sure based on @HaoXuAI's latest changes but it's worth a consideration

Reply:
I think it is embedded within ODFVs, not standalone objects ?